code/Test_definitions/DeviceIdentifier_retrieve_identifier.feature

```diff
@@ -0,0 +1,141 @@
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And one of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When a device_identifier1 is retrieved on the device with SIMCARD1 in slot1
+    And a device_identifier2 is retrieved on the same device with SIMCARD2 in slot1
+    Then device_identifier1 and device_identifier2 are equal
+
+
+  @DeviceIdentifier_retrieve_identifier102_ipv4Address_success
+  Scenario:  retrieve device identifer for phone number PHONENUMBER2, network connection and access token matches PHONENUMBER2
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And one of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When the HTTPS "POST" request is sent
+    And the request body has the field ipv4Address.publicAddress with a value of PUBLICIPV4ADDRESS
+    And the request body has the field ipv4Address.publicPort with a value of PUBLICPORT
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response header "Content-Type" is "application/json"
+    And the response body complies with the OAS schema at "/components/schemas/200RetrieveIdentifier"
+    Then the response status code is 200
+    And the response property "$.imeisv" is IMEISV2
+
+  @DeviceIdentifier_retrieve_identifier200_missing_device_information
+  Scenario:  retrieve identifier but no device information in request
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And one of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When the HTTPS "POST" request is sent
+    And the connection the request is sent over originates from a device with PHONENUMBER1
+    And the request body has NO the field phoneNumber nor other device information
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response header "Content-Type" is "application/json"
+    And the response body complies with the OAS schema at "/components/schemas/ErrorResponse"
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+  @DeviceIdentifier_retrieve_identifier201_missing_scope
+  Scenario:  retrieve device identifier with valid access token but scope device-identifier:retrieve-identifier is missing
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And none of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When the HTTPS "POST" request is sent
+    And the connection the request is sent over originates from a device with PHONENUMBER1
```

Comment:
I'll leave this line in for now, but I don't think this affects the test in any way other than to complicate it. What matters is how the access token is obtained.

This device itself could use client credentials if allowed by regulation, or it could use CIBA if no server involved, or it could use authorisation code flow if a server sent it an appropriate re-direct (in which case, the API call itself would be made by that server and not the device).

All of these options (2-legged token, or 3-legged token obtained using CIBA or authorisation code flow) are also available to a server.